library_management/views.py

```diff
@@ -0,0 +1,164 @@
+from .models import BookRequest
+from .forms import NewTicketForm
+from .models import BookIssue, BookRequest, BookReturn, NewBookTicket
+from .forms import BookReturnForm
+from .models import BookIssue
+from .forms import BookRequestForm
+from django.shortcuts import render, redirect
+from django.views import View
+from .models import Book
+from django.views.generic import ListView
+from django.shortcuts import render
+from .permissions import IsStaffEditorPermission
+from django.urls import reverse_lazy
+from django.views.generic.edit import CreateView
+from .models import NewBookTicket
+from .forms import NewTicketForm
+from rest_framework import generics
+from .models import Book, BookIssue, BookRequest
+from .serializers import BookSerializer, BookIssueSerializer, BookRequestSerializer, BookTicketSerializer
+from django.utils import timezone
+from django.http import HttpResponse
+
+
+class BookList(generics.ListCreateAPIView):
+    queryset = Book.objects.all()
+    serializer_class = BookSerializer
+    permission_classes = [IsStaffEditorPermission]
+
+
+class BookDetail(generics.RetrieveAPIView):
+    queryset = Book.objects.all()
+    serializer_class = BookSerializer
+    lookup_field = 'pk'
+
+
+class BookIssueList(generics.ListCreateAPIView):
+    queryset = BookIssue.objects.all()
+    serializer_class = BookIssueSerializer
+
+
+class BookRequestList (generics.ListAPIView):
+    queryset = BookRequest.objects.all()
+    serializer_class = BookRequestSerializer
+
+
+class BookSearchView(View):
+    template_name = 'book_search_results.html'
+
+    def get(self, request, *args, **kwargs):
+        book_name = self.kwargs['book_name']
+        books = Book.objects.filter(name__icontains=book_name)
+        return render(request, 'book_search_results.html', {'books': books, 'book_name': book_name})
+
+
+class BookRequestView(View):
+    template_name = 'book_request.html'
+
+    def get(self, request):
+        form = BookRequestForm()
+        return render(request, self.template_name, {'form': form})
+
```

Comment:
I don't see permissions being applied here. Shouldn't users and librarian have different permissions, even serializers for this modal?

Reply:
Yes, user and librarian  have separate permissions. I have updated the code.Kindly provide feedback